src/datatable.py

```diff
@@ -0,0 +1,192 @@
+from collections import namedtuple
+from src import mongo
+from querystring_parser import parser
+import json
+
+max_string_length = 50
+
+
+class DataTablesServer:
+
+    def __init__(self, request, columns, index, collection, group_by=None):
+        self.columns = columns
+        self.index = index
+        self.collection = collection
+        self.group_by = group_by
+
+        # values specified by the datatable for filtering, sorting, paging
+        self.request = parser.parse(request.query_string)
+        # print(json.dumps(self.request, sort_keys=True, indent=4))
+
+        self.dbh = mongo.db
+        self.result_data = None  # results from the db
+        self.records_filtered = 0  # total in the table after filtering
+        self.records_total = 0  # total in the table unfiltered
+
+        self.run_queries()
+
+    def output_result_on_given_fields(self):
+        output = {}
+        output['draw'] = str(int(self.request['draw']))
+        output['recordsTotal'] = str(self.records_total)
+        output['recordsFiltered'] = str(self.records_filtered)
+
+        data_rows = []
+        i = self.paging().start
+
+        for row in self.result_data:
+            data_row = []
+            i += 1
+            data_row.append("<a href='/vus/{0}'>{1}</a>".format(row['_id'], i))
+            for col in self.columns:
+                if col in row:
+                    val = row[col]
+                else:
+                    val = "-"
+
+                if isinstance(val, dict):
+                    print("YAS")
+                    val = json.dumps(val)
+                    print(val)
+                elif isinstance(val, list):
+                    val = ", ".join(val)
+
+                if isinstance(val, str):
+                    val = (val[:max_string_length] + '...') if len(val) > max_string_length else val
```

Comment:
I think it is a valid question, however one I cannot answer at this moment.
I do agree that this comment you make is within the scope of this pull request, however I haven't touched or worked this file when developing the new features mentioned in the text I wrote for this pull request.

reasoning:
I haven't anticipated for this to happen. To explain a little bit, the previous programmer (who also left this project completely) did almost everything in the development branch, because there was not a first production version yet. Only some test versions. I had not noticed this on time, so when I opened this release branch to merge everything into the main branch, these code changes were presented to you as a reviewer.

I do want to leave these comments for now, because:
1. This code was validated earlier (see all commits in this pull request by previous programmer)
2. The first production version is now well on its way
3. The current code works correctly
4. I expect a lot of changes based on the 1st production version in scope and functionalities. To have a in-depth look at it now seems a bit overkill.

I have made another issue in github #19, so that the next version needs to have a look.

Reply:
Yes, you are right.  The `if isinstrance(val, str)` is still needed in case `row[col]` is a string.
The `if isinstrance(val, str)` will not be used on `val = "-"`. Since the only thing that happens inside the if statement is stripping the value if it is too long. "-" will never be too long, so the check and stripping is not required.